<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<title>AFAMS: README</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">AFAMS
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.1.2 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li class="current"><a href="pages.html"><span>Related&#160;Pages</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Pages</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title">README </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p>AGOAMS (Au, o matic Gene Ontology Annotation for massive sequencing) GOAA (Gene Ontology Automatic Anotation) AFAMS (Automatic Functional Annotator for Masive Sequencing) AFA (Automatic Functional Anotator)</p>
<h1>GeneOntology-Python</h1>
<p>A set of scripts in python designed to work with Gene Ontology project.</p>
<p>En esta artículo se presenta y evalúa _______________ una herramienta bioinformática libre, está diseñada para automatizar y facilitar el proceso de anotación funcional de genomas y transcriptomas a traves de los terminos de Gene Ontology. Las anotaciones de Gene Ontology son de gran utilidad para entender las funciones asociadas con las secuencias genómicas y transcriptómicas. ______ permite generar asociaciones masivas de términos de Gene Ontology a genomas y transcriptomas. _____ además de hacer una anotación funcional de genomas y transcriptomas es capaz de generar diagramas de fácil interpretación para un usuario a partir de una lista de términos en los cuales quiera centrar su investigación. _____ está compuesto de varios pasos los cuales el usuario puede ejecutar uno a la vez o todos de una vez según sea su necesidad. Al comparar ____ con herramientas como Blast2GO y CateGOrizer se evidenció ______________________.</p>
<h2>Setup </h2>
<h3>Dependences </h3>
<ul>
<li>
Python 2.7 </li>
<li>
MySql Database  </li>
</ul>
<h3>Optional Dependences</h3>
<h4>Graphing tools:</h4>
<ul>
<li>
CairoSVG </li>
<li>
Cairo </li>
<li>
Tinycss  </li>
<li>
Cssselect  </li>
<li>
Pygal </li>
<li>
Pycha </li>
</ul>
<h4>How to install optional dependences? </h4>
<p>easy_install CairoSVG tinycss cssselect pygal <br/>
</p>
<p>Clone the git repository <br/>
</p>
<p>git clone <a href="https://github.com/alejo0317/GeneOntology-Python.git">https://github.com/alejo0317/GeneOntology-Python.git</a></p>
<h3>Configure database </h3>
<p>Edit the config.py file with your database settings (user, password, database)</p>
<h2>Download mappings file and populate mysql database </h2>
<p>Download the file idmapping.tab from <a href="ftp://ftp.pir.georgetown.edu/databases/idmapping/idmapping.tb">ftp://ftp.pir.georgetown.edu/databases/idmapping/idmapping.tb</a> and use it to populate your local database to generate GeneOntology associations. <br/>
</p>
<p>To populate the database with the file idmapping.tab use the script named mappingsToDB.py. This a populate example: <br/>
 python2 mappingsToDb.py /path/to/idmapping.tab</p>
<h2>How to use </h2>
<p>You can use the scripts one by one or can use the full wrapper of all process.</p>
<h2>Scripts on this repository </h2>
<h4>GoDistribution.py </h4>
<p>Generates two files: a file containing relation between GO Terms and sequences, another file containing the tabbed counts of GO terms wanted.</p>
<h4>hits2go.py </h4>
<p>Associantes Uniprot, Refseq, GI, accessions to GO identifiers using a mappings table.</p>
<h4>GraphPie.py </h4>
<p>Generates a Pie chart from a file with the GO counts.</p>
<h2>Test data include </h2>
<p>We have include a set of files to test this scripts. You will find it on the folder test_data</p>
<h4>sequences2hits.csv</h4>
<p>A blast-generated-file with the querys and subjects (NR and Uniprot database has been used) </p>
<h4>sequences2Gos.csv </h4>
<p>A relation between Sequences and his associated GO terms. </p>
<h4>gos2Sequences.csv </h4>
<p>A relation between GO terms and his association with the Sequences. </p>
<h4>gosCounts.tab </h4>
<p>Ready to graph file.</p>
<h2>Other files include </h2>
<h4>go.obo </h4>
<p>This file contain s the descriptions and relations bettwen the avaliable GO terms. Has been downloaded from <a href="http://purl.obolibrary.org/obo/go.obo">http://purl.obolibrary.org/obo/go.obo</a> </p>
<h4>Celullar_Component </h4>
<p>A list of "2nd level" GO terms from the GO category Celullar Component (GO:0005575) </p>
<h4>Biological_Process </h4>
<p>A list of "2nd level" GO terms from the GO category Biological Process (GO:0008150) </p>
<h4>Mollecular_Function </h4>
<p>A list of "2nd level" GO terms from the GO category Mollecular Function (GO:0003674) </p>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Fri Nov 22 2013 08:08:35 for AFAMS by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.1.2
</small></address>
</body>
</html>
